### PR TITLE
engine: add configurations for filter enhancements

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Ref #12842 

```commit-message
Add some configurations for RocksDB filter enhancements
```

### Related changes

https://github.com/tikv/rust-rocksdb/pull/740

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests
- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -547,16 +559,24 @@ macro_rules! build_cf_opt {
         block_base_opts
             .set_pin_l0_filter_and_index_blocks_in_cache($opt.pin_l0_filter_and_index_blocks);
         if $opt.use_bloom_filter {
-            block_base_opts.set_bloom_filter(
-                $opt.bloom_filter_bits_per_key as f64,
-                $opt.block_based_bloom_filter,
-            );
+            if let Some(level) = $opt.ribbon_filter_above_level {
```
Comment (tonyxuqqi) [inline src/config/mod.rs:562]: what if the level value exceeds 6.

**Review by tonyxuqqi — APPROVED** [review]

File: src/config/mod.rs (hunk 1 above)
Comment (tabokie) [reply src/config/mod.rs:562]: Then it's effectively disabled.

**Review by SpadeA-Tang — APPROVED** [review]

**Pull request merged**